pgdrive/scene_manager/agent_manager.py

```diff
@@ -129,42 +131,36 @@ def _check(self):
 
     def propose_new_vehicle(self):
         self._check()
-        if len(self.__pending_objects) > 0:
-            obj_name = list(self.__pending_objects.keys())[0]
-            self._check()
-            v = self.__pending_objects.pop(obj_name)
-            v.prepare_step([0, -1])
-            v.chassis_np.node().setStatic(False)
-            self.observations[obj_name].reset(v)
-            return self.allow_respawn, dict(
-                vehicle=v,
-                observation=self.observations[obj_name],
-                observation_space=self.observation_spaces[obj_name],
-                action_space=self.action_spaces[obj_name],
-                old_name=self.__object_to_agent[obj_name],
-                new_name="agent{}".format(self.next_agent_count)
-            )
-        return None, None
-
-    def confirm_respawn(self, success: bool, vehicle_info):
-        vehicle = vehicle_info['vehicle']
-        if success:
-            vehicle.set_static(False)
-            self.next_agent_count += 1
-            self.__active_objects[vehicle.name] = vehicle
-            self.__object_to_agent[vehicle.name] = vehicle_info["new_name"]
-            self.__agent_to_object.pop(vehicle_info["old_name"])
-            self.__agent_to_object[vehicle_info["new_name"]] = vehicle.name
-        else:
-            vehicle.set_static(True)
-            self.__pending_objects[vehicle.name] = vehicle
+        obj_name = list(self.__pending_objects.keys())[0]
         self._check()
+        v = self.__pending_objects.pop(obj_name)
+        v.prepare_step([0, -1])
+        self.observations[obj_name].reset(v)
+        new_agent_id = self.next_agent_id()
+        dead_vehicle_id = self.__object_to_agent[obj_name]
+        vehicle = v
```

Comment:
don't use `v` and `vehicle` both here! Please clean the variable name.

Reply:
fixed